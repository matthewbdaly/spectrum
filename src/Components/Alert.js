// @flow
import React from 'react';
import type { Node } from 'react';
import './Alert.scss';

type Props = {
  theme: string,
  children?: Node,
  dismissable: boolean
};

const Alert = (props: Props) => {
  let theme = 'alert' + (props.theme ? (' ' + props.theme) : '');
  return (
    <div className={theme}>
      {props.children}
    </div>
  );
}

export default Alert;
