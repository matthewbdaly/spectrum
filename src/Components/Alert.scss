@import "../variables";

div.alert {
    width: 100%;
    background-color: $background;
    color: $foreground;
    border-color: $foreground;
    border-width: $btn-border-width;
    border-style: $btn-border-style;
    margin: $btn-margin;
    border-radius: $btn-radius;
    padding: $btn-padding;

    &.main {
        background-color: $theme-main;
        color: $background;
        border-color: $theme-main;

        &.inverse {
            color: $theme-main;
            background-color: $background;
        }
    }

    &.positive {
        background-color: $theme-positive;
        color: $background;
        border-color: $theme-positive;

        &.inverse {
            color: $theme-positive;
            background-color: $background;
        }
    }

    &.negative {
        background-color: $theme-negative;
        color: $background;
        border-color: $theme-negative;

        &.inverse {
            color: $theme-negative;
            background-color: $background;
        }
    }


}
