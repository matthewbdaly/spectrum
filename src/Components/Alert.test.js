import React from 'react';
import ReactDOM from 'react-dom';
import Alert from './Alert';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Alert />, div);
});

it('should match the snapshot', () => {
  const div = document.createElement('div');
  const el = ReactDOM.render(<Alert />, div);
  expect(el).toMatchSnapshot();
});
