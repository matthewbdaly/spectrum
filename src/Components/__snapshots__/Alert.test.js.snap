// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Alert {
  "_reactInternalFiber": FiberNode {
    "_debugID": 8,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "<PROJECT_ROOT>/src/Components/Alert.test.js",
      "lineNumber": 12,
    },
    "alternate": null,
    "child": FiberNode {
      "_debugID": 9,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "<PROJECT_ROOT>/src/Components/Alert.js",
        "lineNumber": 25,
      },
      "alternate": null,
      "child": null,
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": undefined,
        "className": "alert",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": undefined,
        "className": "alert",
      },
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <div
        class="alert"
/>,
      "tag": 5,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": Object {
      "theme": "alert",
    },
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugID": 6,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 6,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
                        class="alert"
            />
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "capturedValues": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Alert />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Alert />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 32,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Alert />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
                    class="alert"
          />
</div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <Alert />,
        },
        "callbackList": null,
        "capturedValues": null,
        "expirationTime": 0,
        "first": null,
        "hasForceUpdate": false,
        "isInitialized": true,
        "isProcessing": false,
        "last": null,
      },
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": Object {
    "theme": "alert",
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
